google/cloud/firestore_v1/base_vector_query.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2024 Google LLC All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Classes for representing vector queries for the Google Cloud Firestore API.
+"""
+
+import abc
+
+from abc import ABC
+from enum import Enum
+from typing import Iterable, Optional, Tuple, Union
+from google.api_core import gapic_v1
+from google.api_core import retry as retries
+from google.cloud.firestore_v1.base_document import DocumentSnapshot
+from google.cloud.firestore_v1.types import query
+from google.cloud.firestore_v1.vector import Vector
+from google.cloud.firestore_v1 import _helpers
+
+
+class DistanceMeasure(Enum):
+    EUCLIDEAN = 1
+    COSINE = 2
+    DOT_PRODUCT = 3
+
+
+class BaseVectorQuery(ABC):
+    """Represents a vector query to the Firestore API."""
+
+    def __init__(self, nested_query) -> None:
+        self._nested_query = nested_query
```

Comment:
Added an arg comment from vector_query. Not sure if Type hinting is the right thing to do since it introduces circular dependency for import (looks like aggregation avoids it for the similar reason)..